### PR TITLE
update ZH stats to 2M events, lower lr

### PR DESCRIPTION
**Pull request by jpata**
- due to the overtraining observed in the dm_multiclass and jet_regression tasks, bump the signal statistics
- retraining ongoing into `240522_qq_zh_2m_merged`
- training time <10h for jet_regression and dm_multiclass (on 10g MIG partition on A100)
- use cosine learning rate decay and a lower learning rate (1e-4)
- clip `max_cands` to 16 (fine for signal, but to be increased to 32 for background)
- the final results are in https://github.com/HEP-KBFI/ml-tau-en-reg/pull/21#issuecomment-2132970791

### CODE REVIEW

**Comment by jpata** [comment]
In [2a89e4c](https://github.com/HEP-KBFI/ml-tau-en-reg/pull/21/commits/2a89e4cf082af0438e59421d88a3964d4204b659) I introduced cosine learning rate decay, a lower learning rate, and a higher batch size.

The jet_regression loss in now better behaved for ParticleTransformer:
Screenshot 2024-05-24 at 19 09 36

But the dm_multiclass validation loss is still fluctuating quite wildly for ParticleTransformer: 
Screenshot 2024-05-24 at 19 10 13

**Comment by jpata** [comment]
The training results are at `/local/joosep/ml-tau-en-reg/results/240524_cosinescheduler/`.
With early stopping (10 epochs), mostly the models finished within <10h.

CM_zh
median_and_iqr
[loss_dm_multiclass.pdf](https://github.com/HEP-KBFI/ml-tau-en-reg/files/15453243/loss_dm_multiclass.pdf)
[loss_jet_regression.pdf](https://github.com/HEP-KBFI/ml-tau-en-reg/files/15453244/loss_jet_regression.pdf)

File: enreg/config/model_training.yaml
Comment (Laurits7) [inline enreg/config/model_training.yaml:1]: Is max_cands change from 25 -> 16 reasonable? On tau samples 16 should be fine, but on qq samples probably the number is too low. Although currently binary classifier for tau ID is not a priority, then changing it back and forth in the future might not be something we want.

image

**Review by Laurits7 — APPROVED** [review]
Other than the number of max_cands, everything seems to be in order.
